pkg/bloomgateway/client.go

```diff
@@ -204,6 +205,51 @@ func (c *GatewayClient) Close() {
 	c.dnsProvider.Stop()
 }
 
+func (c *GatewayClient) PrefetchBloomBlocks(ctx context.Context, blocks []bloomshipper.BlockRef) error {
+	if len(blocks) == 0 {
+		return nil
+	}
+
+	pos := make(map[string]int)
+	servers := make([]addrWithBlocks, 0)
```

Comment:
allocate for worst case:

```suggestion
	servers := make([]addrWithBlocks, 0, len(blocks))
```